torchvision/transforms/functional_tensor.py

```diff
@@ -48,3 +48,25 @@ def crop(img, top, left, height, width):
         raise TypeError('tensor is not a torch image.')
 
     return img[..., top:top + height, left:left + width]
+
+def to_grayscale(img, num_output_channels = 3):
+    """Convert the given RGB Image Tensor to Grayscale.
+
+    Args
+        img (Tensor): Image to be converted to Grayscale in the form [C, H, W].
+        num_output_channels (int): denotes the number of channels to return after conversion
+    Returns:
+    Tensor: Grayscale image.
+    """
+    if not F._is_tensor_image(img):
+        raise TypeError('tensor is not a torch image.')
+
+    else:
+        hwc_img = img.transpose(1, 2).transpose(0, 2)
+        weights = torch.tensor([0.2989, 0.5870, 0.1140])
+        res = hwc_img[:,:,:3] * weights[None, :]
```

Comment:
It might be more efficient (specially on the GPU) to do something like
```python
res = torch.tensordot(img, weights[:, None, None], [[0], [0]]).squeeze()
```
although this might be a bit less readable. Can you do some benchmarks to see if there is any difference and report back?